### PR TITLE
Cleaning time, x1.

### PR DESCRIPTION
**Pull request by tigercat2000**
Mob Changes:
 - Human Bump() code cleansed.
  - Drones have been unsnowflaked from the Bump(), they are now density 0.
    Happy side effect; they can walk under people now.
 - Human attack_alien() code cleansed.
  - Mostly shitty insane checks.
 - Human damage/husking code cleansed.
  - More shitty insane if statements.
 - Human retarded speech moved out of insanely long pick() proc, made into
   lists.

Gamemode Changes:
 - Enemy intercept (send_intercept()) completely nuked, it did not compile
   when I attempted to make it into a config option.
  - The intercept for blob() is retained and made into a custom proc for
    blob.
 - The constant variables 'waittime_l' and 'waittime_h' have been moved to
   /datum/game_mode, as they were defined on every gamemode with the same
   values.
 - rp-revolution.dm and anti_revolution.dm deleted. They do not compile,
   and are not included in the DME already.

I noticed a shitload of insane procs while trying to figure out #2575 (still no fucking idea), so I cleaned them up. Then noticed gamemodes had horrifying messy things too. And cleaned them.
If you know of anything especially bad that needs to be cleaned, I might add it to this.

### CODE REVIEW

**Comment by tigercat2000** [comment]
makeDatumRefList changelog:
Changelog:
 - Switched surgeries to use init_subtypes()
 - Updated init_subtypes()
 - Made the species race_key handling a bit less line-taking (uses
   pre-increment now)
 - Added new proc, init_datum_subtypes, which takes 4 arguments-
   init_datum_subtypes(prototype, list/L, list/pexempt, assocvar)

   prototype is a path which will be run through subtypesof
   list/L is the list that new datums will be added to
   pexempt is subtracted from the 'subtypesof(prototype)' (exempting them)
   assocvar is the var to associate by in the list- uses vars[] lookup

   Jobs, superheroes, and languages now use this proc.

**Comment by Keilden** [comment]
Sweet no longer do I need to try to scare away people who are in my way with my scary buzz sound.

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 1]
```diff
@@ -143,34 +149,39 @@
...
-	var/mut_prob = min(80, getCloneLoss()+10)
-	if (amount > 0)
-		if (prob(mut_prob))
-			var/list/obj/item/organ/external/candidates = list()
-			for (var/obj/item/organ/external/O in organs)
-				if(O.status & ORGAN_ROBOT) continue
+	var/mut_prob = min(80, getCloneLoss() + 10)
+	if(amount > 0) //cloneloss is being added
+		if(prob(mut_prob))
+			var/list/obj/item/organ/external/candidates = list() //TYPECASTED LISTS ARE NOT A FUCKING THING WHAT THE FUCK
```
Comment (RemieRichards) [inline code/modules/mob/living/carbon/human/human_damage.dm]: this is only for decoration, p.sure.
a way to say "this is a list of limbs"
granted you can only tell here at the definition, where as a new name like ```candidate_limbs``` lets you figure it out wherever you see the var.

tl;dr they do, but it's for decoration.

Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/human_damage.dm]: I think this is best refered to as "grep protection"
